Sources/JWTKit/JWTSigners.swift

```diff
@@ -280,3 +295,231 @@ private struct JWKSigner {
+    /// - Throws: An error if the token cannot be verified or decoded.
+    /// - Returns: The verified and decoded payload of the specified type.
+    @available(*, deprecated, message: "Please make sure Payload conforms to AsyncJWTPayload instead of JWTPayload before updating to v5.")
+    public func verify<Payload, Data: DataProtocol&Sendable>(
+        _ token: Data,
+        as _: Payload.Type = Payload.self
+    ) async throws -> Payload
+    where Payload: JWTPayload {
+        try signers.verify(token, as: Payload.self)
+    }
+
+    /// Verifies and decodes a JWT token to extract the payload.
+    ///
+    /// - Parameters:
+    ///   - token: A JWT token string.
+    ///   - as: The type of payload to decode.
+    ///   - iteratingKeys: Whether to try verifying the token with all keys in the collection.
+    /// - Throws: An error if the token cannot be verified or decoded.
+    /// - Returns: The verified and decoded payload of the specified type.
+    public func verify<Payload>(
+        _ token: String,
+        as _: Payload.Type = Payload.self,
+        iteratingKeys: Bool = false
+    ) async throws -> Payload
+    where Payload: AsyncJWTPayload {
+        try await verify(Array(token.utf8), as: Payload.self)
+    }
+
+    /// Verifies and decodes a JWT token to extract the payload.
+    ///
+    /// - Parameters:
+    ///   - token: A JWT token.
+    ///   - as: The type of payload to decode.
+    ///   - iteratingKeys: Whether to try verifying the token with all keys in the collection.
+    /// - Throws: An error if the token cannot be verified or decoded.
+    /// - Returns: The verified and decoded payload of the specified type.
+    public func verify<Payload, Data: DataProtocol&Sendable>(
+        _ token: Data,
+        as _: Payload.Type = Payload.self
+    ) async throws -> Payload
+    where Payload: AsyncJWTPayload {
+        let parser = try JWTParser(token: token)
+        let header = try parser.header(jsonDecoder: signers.defaultJSONDecoder)
+        let signer = try signers.require(kid: header.kid, alg: header.alg)
+        return try await signer.verify(parser: parser)
+    }
+    
+    /// Signs a JWT payload and returns the JWT string.
+    ///
+    /// - Parameters:
+    ///   - payload: The payload to sign.
+    ///   - kid: An optional key identifier to specify the signer.
+    ///         If not provided, the header is checked for a KID,
+    ///         and if that is not provided, the default signer is used.
+    ///   - header: An optional header to include in the JWT.
+    /// - Throws: An error if the payload cannot be signed.
+    /// - Returns: A signed JWT token string.
+    @available(*, deprecated, message: "Please make sure Payload conforms to AsyncJWTPayload instead of JWTPayload before updating to v5.")
+    public func sign<Payload: JWTPayload>(
+        _ payload: Payload,
+        kid: JWKIdentifier? = nil
+    ) async throws -> String {
+        try signers.sign(payload, kid: kid)
+    }
```

Comment:
This variant is marked as deprecated despite appearing on the new JWTKeyCollection to help push users to adopt AsyncJWTPayload wherever they use another payload type.